csharp/src/Ice/SignaledSocketStream.cs

```diff
@@ -13,28 +13,33 @@ namespace ZeroC.Ice
     /// for receiving data. The socket can easily signal the stream when new data is available.</summary>
     internal abstract class SignaledSocketStream<T> : SocketStream, IValueTaskSource<T>
     {
-        internal bool IsSignaled => _source.GetStatus(_source.Version) != ValueTaskSourceStatus.Pending;
+        internal bool IsSignaled => Thread.VolatileRead(ref _signaled) == 1;
```

Comment:
This is not equivalent to marking _signaled volatile?

Reply:
It is. I don't want _signaled to be a volatile here as it's being incremented/decremented with Interlocked which doesn't need a volatile.